### PR TITLE
Added a section detailing toxcore's state format

### PR DESCRIPTION
**Pull request by alexbakker**
Because I had to reverse engineer this from toxcore's code, it's possible that I missed something.
@irungentoo, please review this and let me know if anything is incorrect.

[Review on Reviewable](https://reviewable.io/reviews/toktok/tox-spec/26)

### CODE REVIEW

**Comment by iphydf** [comment]
Suggested diff against your change (formatting): https://gist.github.com/iphydf/062c3fdb7e553532be6e

File: spec.md [hunk 1]
```diff
@@ -3713,3 +3713,163 @@ that was stored and it is removed from the array.
...
+
+### Nospam and Keys
+
+Length | Contents
+------ | --------
+`4`    | `uint32_t` Nospam
+`32`   | Long term public key
+`32`   | Long term secret key
+
+### DHT
```
Comment (iphydf) [inline spec.md]: MD024 Multiple headers with the same content

Please choose a different name or qualify it (or the existing one).

File: spec.md [hunk 2]
```diff
@@ -3713,3 +3713,163 @@ that was stored and it is removed from the array.
...
+
+This section contains a list of DHT-related sections.
+
+Length   | Contents
+-------- | --------
+`4`      | `uint32_t` (0x159000D)
+`?`      | List of DHT sections
+
+#### Sections
```
Comment (iphydf) [inline spec.md]: MD024 Multiple headers with the same content

"Sections" exists twice in this document.

**Comment by alexbakker** [comment]
Patch applied and issues fixed. 
`make check` shouldn't fail anymore.

**Comment by iphydf** [comment]
:lgtm: waiting for @irungentoo to review for correctness.

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/toktok/tox-spec/26#-:-KCKknKPMCcHId1mGJCZ:-1092339110)_

**Comment by iphydf** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

_[spec.md, line 3728 [r2]](https://reviewable.io:443/reviews/toktok/tox-spec/26#-KCKmcTiqUdM30PQz0xt:-KCKmcTiqUdM30PQz0xu:1873063320) ([raw file](https://github.com/toktok/tox-spec/blob/1d1de2a98dcbd7880d19d3099bd3ad837688b5d1/spec.md#L3728)):_
Can you change the table format like this:

``` diff
-|--------|-----------------------------------|
+|:-------|:----------------------------------|
```

My patch was slightly wrong. The canonical table format in the document is with `|:` as cell separator.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/toktok/tox-spec/26)_

**Comment by zetok** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

_[spec.md, line 3731 [r1]](https://reviewable.io:443/reviews/toktok/tox-spec/26#-KCGVC2DKfKYOHKU4QRW:-KCGVC2DKfKYOHKU4QRX:-1315336286) ([raw file](https://github.com/toktok/tox-spec/blob/ebfc3f6c1b67efb5d877b3e18d4ce9494e13f8ad/spec.md#L3731)):_
perhaps s/List of sections/Sections/ ?

I.e. "list" sounds as if that field contained just some sort of list, rather than whole sections.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/toktok/tox-spec/26)_

**Comment by alexbakker** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

_[spec.md, line 3731 [r1]](https://reviewable.io:443/reviews/toktok/tox-spec/26#-KCGVC2DKfKYOHKU4QRW:-KCLBdxmPQ8httig-wJQ:1107563104) ([raw file](https://github.com/toktok/tox-spec/blob/ebfc3f6c1b67efb5d877b3e18d4ce9494e13f8ad/spec.md#L3731)):_
I disagree. Your suggestion makes it less clear in my opinion.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/toktok/tox-spec/26)_

**Comment by alexbakker** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

_[spec.md, line 3731 [r1]](https://reviewable.io:443/reviews/toktok/tox-spec/26#-KCGVC2DKfKYOHKU4QRW:-KCLEk6ULCiQD7Ks8Jp1:-499146009) ([raw file](https://github.com/toktok/tox-spec/blob/ebfc3f6c1b67efb5d877b3e18d4ce9494e13f8ad/spec.md#L3731)):_
Done.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/toktok/tox-spec/26)_

**Comment by alexbakker** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

_[spec.md, line 3731 [r1]](https://reviewable.io:443/reviews/toktok/tox-spec/26#-KCGVC2DKfKYOHKU4QRW:-KCLF7rm5seNsW_pmOdE:810683252) ([raw file](https://github.com/toktok/tox-spec/blob/ebfc3f6c1b67efb5d877b3e18d4ce9494e13f8ad/spec.md#L3731)):_
Okay, reviewable is garbage. Ignore that last message.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/toktok/tox-spec/26)_

**Comment by alexbakker** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

_[spec.md, line 3728 [r2]](https://reviewable.io:443/reviews/toktok/tox-spec/26#-KCKmcTiqUdM30PQz0xt:-KCLFKXb1TK7FkCYFPbw:-499146009) ([raw file](https://github.com/toktok/tox-spec/blob/1d1de2a98dcbd7880d19d3099bd3ad837688b5d1/spec.md#L3728)):_
Done.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/toktok/tox-spec/26)_

**Comment by iphydf** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/toktok/tox-spec/26)_

**Comment by iphydf** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_[spec.md, line 3731 [r1]](https://reviewable.io:443/reviews/toktok/tox-spec/26#-KCGVC2DKfKYOHKU4QRW:-KCkqeoIgl2Dz43cgNFc:-263000966) ([raw file](https://github.com/toktok/tox-spec/blob/ebfc3f6c1b67efb5d877b3e18d4ce9494e13f8ad/spec.md#L3731)):_
Let's review this later.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/toktok/tox-spec/26)_

**Comment by iphydf** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/toktok/tox-spec/26)_

**Pull request merged**